types/src/transaction/mod.rs

```diff
@@ -2620,3 +2632,52 @@ impl ViewFunctionOutput {
         Self { values, gas_used }
     }
 }
+
+#[derive(Debug, Clone, Copy)]
+pub struct ExtraInfo {
+    persisted_info: PersistedExtraInfo,
+    ephemeral_info: Option<EphemeralExtraInfo>,
+}
+
+impl ExtraInfo {
+    pub fn new(
+        persisted_info: PersistedExtraInfo,
+        ephemeral_info: Option<EphemeralExtraInfo>,
+    ) -> Self {
+        Self {
+            persisted_info,
+            ephemeral_info,
+        }
+    }
+
+    pub fn new_empty() -> Self {
+        Self {
+            persisted_info: PersistedExtraInfo::None,
+            ephemeral_info: None,
+        }
+    }
+
+    pub fn into_persisted_info(self) -> PersistedExtraInfo {
+        self.persisted_info
+    }
+
+    pub fn persisted_info(&self) -> &PersistedExtraInfo {
+        &self.persisted_info
+    }
+
+    pub fn ephemeral_info(&self) -> &Option<EphemeralExtraInfo> {
+        &self.ephemeral_info
+    }
+}
+
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+pub enum PersistedExtraInfo {
+    None,
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct EphemeralExtraInfo {
+    // TODO(grao): After execution pool is implemented we might want this information be persisted
+    // onchain?
+    pub proposer_index: u64,
```

Comment:
Doesn't state sync need this information again? State sync can't derive this information just from the downloaded transactions chunks right?

Reply:
EphemeralExtraInfo is intend for information that is not needed in VM.
I'm using this information to generate the input of BlockEpilogue txn on the consensus path for now. On the state sync path we only need the input of BlockEpilogue txn.